File: medplum-java/medplum-java-model/src/main/java/com/fhir/Coverage.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCoverage.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Coverage {
  Reference beneficiary();

  Optional<Period> period();

  Optional<Reference> subscriber();

  Optional<Reference> policyHolder();

  Optional<String> subscriberId();

  Optional<Meta> meta();

  Optional<List<Coverage_CostToBeneficiary>> costToBeneficiary();

  Optional<CodeableConcept> relationship();

  Optional<String> network();

  Optional<List<Extension>> extension();

  Optional<id> id();

  String resourceType();

  Optional<code> status();

  Optional<Boolean> subrogation();

  @JsonProperty("class")
  Optional<List<Coverage_Class>> _class();

  Optional<uri> implicitRules();

  Optional<List<Extension>> modifierExtension();

  List<Reference> payor();

  Optional<CodeableConcept> type();

  Optional<Narrative> text();

  Optional<List<Reference>> contract();

  Optional<positiveInt> order();

  Optional<code> language();

  Optional<String> dependent();

  Optional<List<Identifier>> identifier();

  Optional<List<ResourceList>> contained();

  static ImmutableCoverage.BeneficiaryBuildStage builder() {
    return ImmutableCoverage.builder();
  }
}

File: medplum-typescript/src/model/fhir/Coverage.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Coverage_Class } from "./Coverage_Class";
import { Coverage_CostToBeneficiary } from "./Coverage_CostToBeneficiary";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { positiveInt } from "./positiveInt";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface Coverage {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    status: code | null | undefined;
    type: CodeableConcept | null | undefined;
    policyHolder: Reference | null | undefined;
    subscriber: Reference | null | undefined;
    subscriberId: string | null | undefined;
    beneficiary: Reference;
    dependent: string | null | undefined;
    relationship: CodeableConcept | null | undefined;
    period: Period | null | undefined;
    payor: Reference[];
    class: Coverage_Class[] | null | undefined;
    order: positiveInt | null | undefined;
    network: string | null | undefined;
    costToBeneficiary: Coverage_CostToBeneficiary[] | null | undefined;
    subrogation: boolean | null | undefined;
    contract: Reference[] | null | undefined;
}
